src/views/received_sms_list/Filter.js

```diff
@@ -1,78 +1,133 @@
-import React from 'react'
+import React, { useState, useCallback, useEffect } from 'react'
 import { useHistory } from 'react-router-dom'
 import { PropTypes } from '@dhis2/prop-types'
 import i18n from '@dhis2/d2-i18n'
+import debounce from 'lodash.debounce'
 import { dataTest } from '../../dataTest'
 import {
     Button,
-    InputFieldFF,
-    ReactFinalForm,
-    SingleSelectFieldFF,
+    SingleSelectField,
+    SingleSelectOption,
+    InputField,
+    CircularLoader,
 } from '@dhis2/ui'
-import { STATUS_FILTER_OPTIONS } from './config'
-import { useQueryParams } from './useQueryParams'
-import { createSearchString } from './createSearchString'
+import { useQueryParams } from '../../hooks'
+import { createSearchString } from '../../utils'
+import { statusMap } from './translations'
 import styles from './Filter.module.css'
 
-const { Form, Field } = ReactFinalForm
+// From https://github.com/dhis2/dhis2-core/blob/ea76fa86409613c9766d4508d65c88ac55c413c3/dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/incoming/SmsMessageStatus.java
+const STATUS_FILTER_OPTIONS = [
+    'ALL',
+    'FAILED',
+    'INCOMING',
+    'PROCESSED',
+    'PROCESSING',
+    'SENT',
+    'UNHANDLED',
+].map(status => ({ value: status, label: statusMap[status] }))
 
-const Filter = ({ loading }) => {
+const PhoneInputField = ({ phoneNumber, onChange }) => {
+    const [value, setValue] = useState(phoneNumber)
+    const [waiting, setWaiting] = useState(false)
+    useEffect(() => {
+        setValue(phoneNumber)
+    }, [phoneNumber])
+    const debouncedOnChange = useCallback(
+        debounce(event => {
+            onChange(event)
+            setWaiting(false)
+        }, 300),
+        [onChange]
+    )
+    const handleChange = ({ value }) => {
+        setValue(value)
+        setWaiting(true)
+        debouncedOnChange({ value })
+    }
```

Comment:
I've sent you (Medi) a message on slack with regards to our handling of state in query params, would be nice to get rid of the multiple sources of truth and simplify logic like this a bit.